src/flowsynth.py

```diff
@@ -613,11 +613,19 @@ def render(self, eventid):
             else:
                 lyr_ip = IPv6(src = src_host, dst = dst_host)
             lyr_eth = Ether(src = src_mac, dst = dst_mac)
+            # the 'payload' variable is type str [sic] so here we put the payload into a bytearray
+            # and pass it to scapy using Raw(). If we don't do this, we can end up with situations
+            # where bytes (e.g. entered with hex encoding) can get interpreted as multi-byte UTF8
+            # when passed to scapy which is not the desired behavior.
+            payload_bytes = bytearray()
+            payload_bytes.extend(map(ord,payload))
             if (self.l4_proto == Flow.PROTO_UDP):
                 #generate udp packet
-                lyr_udp = UDP(sport = src_port, dport = dst_port) / payload
+                lyr_udp = UDP(sport = src_port, dport = dst_port) / Raw(payload_bytes)
```

Comment:
Be aware the division operator works differently in python2 and python3:

```
▶ python2

WARNING: Python 2.7 is not recommended.
This version is included in macOS for compatibility with legacy software.
Future versions of macOS will not include Python 2.7.
Instead, it is recommended that you transition to using 'python3' from within Terminal.

Python 2.7.16 (default, Jul  5 2020, 02:24:03)
[GCC 4.2.1 Compatible Apple LLVM 11.0.3 (clang-1103.0.29.21) (-macos10.15-objc- on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> 2 / 3
0
>>> ^D

git/attackerdb-deploy/ansible  fix-directories ✗                                                                                                                                                                       3d15h ⚑ ✚ ◒
▶ python3
Python 3.6.10 (default, Apr 28 2020, 09:40:26)
[GCC 4.2.1 Compatible Apple LLVM 11.0.3 (clang-1103.0.32.59)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> 2 / 3
0.6666666666666666
```

If you want an integer (truncated) use `//` which will do the same in both environments.

Reply:
I /think/ this is not real division and is instead a method call disguised by operator overloading.

Reply:
Yeah, it isn't division, it's a scapy syntax thing.